### ISSUE TITLE
Not able to connect the remote worker

### ISSUE THREAD

**Issue by ramprasath-raja** [prdn/pigato#37]
Hi,

We are trying to implement the micro service concept in our project using Node Js. We are using Pigato 0.0.38 for this. The client and worker concept work nice when workers, broker and client in single machine.  During this time, I used the host as "tcp://127.0.0.1:3008" . 

I am trying to implement the same in different server which means client and broker are hosted in one server and workers are hosted in different server. I am using Amazon free tier. In this case, I have changed the Amazon IP. But It is not woking properly.

Find the bellow code where I give broker ip and initiate worker class.

var Worker = require('pigato').Worker;
var worker = new Worker(tcp://brokerHostingIp:3008, 'workername');

Basically, I would like to know whether this plugin supports that workers are hosted in different server.
If you say yes, then Please let me know what I missed.

Find the below my server details.

Node version : 0.12.2
NPM version :  2.7.4
AWS Amazon linux  : 3.14.35-28.38.amzn1.x86_64

Thanks
RamP

**Comment by leonpegg** [comment]
Hi @ramprasath-raja yes Pigato most certainly supports workers,brokers & clients on different machines.

Currently my company uses Pigato on AWS over several machines and even have remote workers located outside AWS (these connect via a proxy using ZAP Curve for authentication)

**Comment by prdn** [comment]
Hi @ramprasath-raja 
I use pigato in both local and distributed environments, even on geographical scale.

Can you try to deploy the example (echo) in the examples directory and change only the IPs and see if that works?
